File: app/src/main/java/com/ed/redditapp/ui/imageview/ImageDownloadAsyncTask.java
package com.ed.redditapp.ui.imageview;

import android.graphics.Bitmap;

import com.ed.redditapp.lib.AsyncTask;
import com.ed.redditapp.lib.http.HttpClient;

public class ImageDownloadAsyncTask {
    public interface Callback {
        void callback(Bitmap image);
    }

    private final HttpClient httpClient;

    public ImageDownloadAsyncTask() {
        httpClient = new HttpClient();
    }

    public void run(String url, Callback callback) {
        new AsyncTask().runAsync(h -> {
            Bitmap bitmap = httpClient.getBitmap(url);
            h.onUIThread(() -> callback.callback(bitmap));
        });
    }
}

File: app/src/main/java/com/ed/redditapp/ui/imageview/ImageDownloadAsyncTask.kt
package com.ed.redditapp.ui.imageview

import android.graphics.Bitmap
import com.ed.redditapp.lib.AsyncTask
import com.ed.redditapp.lib.http.HttpClient

typealias ImageDownloadCallback = (image: Bitmap) -> Unit

class ImageDownloadAsyncTask {
    private val httpClient = HttpClient()

    fun run(url: String, callback: ImageDownloadCallback) {
        AsyncTask().runAsync { h ->
            val bitmap = httpClient.getBitmap(url)
            h.onUIThread { callback(bitmap) }
        }
    }
}